### PR TITLE
Initialize subprojects and command-line interface

### LINKED ISSUE #2: Prepare a native command-line interface for save framework
Now it can have only `--help` option

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Added subprojects
* Added CLI runner
* Added core classes
* Added versions and talaiot gradle plugins

Closes #2

### CODE REVIEW

File: save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import kotlinx.cli.default
+
+/**
+ * @param args CLI args
+ * @return an instance of [SaveConfig]
+ */
+fun createConfigFromArgs(args: Array<String>): SaveConfig {
+    val parser = ArgParser("save")
+
+    val configPath by parser.option(
```
Comment (akuleshov7) [inline save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt:23]: just "config", cli should not have complex argument names. Also camel case is not applicable for cli

File: save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        ArgType.String,
+        shortName = "f",
```
Comment (akuleshov7) [inline save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt:25]: -c

File: save-core/src/commonMain/kotlin/org/cqfn/save/core/config/ReportType.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.cqfn.save.core.config
+
+/**
+ * Possible types of output formats.
+ */
+enum class ReportType {
+    JSON, XML
```
Comment (akuleshov7) [inline save-core/src/commonMain/kotlin/org/cqfn/save/core/config/ReportType.kt:7]: TOML 
PLAIN

File: save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+
+/**
+ * @param args CLI args
+ * @return an instance of [SaveConfig]
+ */
+fun createConfigFromArgs(args: Array<String>): SaveConfig {
```
Comment (akuleshov7) [inline save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt:20]: may be will add some argument for a baseline? (while I haven't yet forgotten about that)
We should have a baseline functionality

**Review by akuleshov7 — CHANGES_REQUESTED** [review]
please see comments and we will merge it as a basic template concept

File: save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt (hunk 4 above)
Comment (petertrr) [reply save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt:21]: Added this and addressed other comments, please take a look

**Review by akuleshov7 — APPROVED** [review]
lgtm

**Pull request merged**